resources/Spelunky2Entities.json

```diff
@@ -2192,16 +2198,26 @@
     ],
     "Robot": [
       { "field": "sound", "type": "SoundMeta" },
-      { "field": "emitted_light_explosion", "type": "IlluminationPointer" }
+      { "field": "emitted_light_explosion", "type": "IlluminationPointer" },
+	  {
+        "field": "prev_move_state",
+        "type": "UnsignedByte",
+        "comment": "used to return to the correct state when landing after falling when activated"
+      }
     ],
     "Quillback": [
       { "field": "sound", "type": "SoundMeta" },
       { "field": "particle", "type": "ParticleEmitterInfoPointer" },
       {
         "field": "unknown_state",
-        "type": "Flags16",
+        "type": "Flags8",
         "comment": "accessed as byte and word, so, flags?, changed mostly when breaking floors"
       },
+      {
+        "field": "hit_wall_direction",
+        "type": "Byte",
+        "comment": "set when in apply_movement to 1 when colliding with a hard wall to the left, -1 if to the right, 0 if no collision."
```

Comment:
I think the variable is more like "post_hit_wall_direction", as direction in which he should face after the collision animation is done. (comparing to movex)

Can you change in the comment `hard wall` to `indestructible wall`, since it also works if you use generic floor with the indestructible flag

also `unknown_state` is probably just "bool destroying_wall". `apply_movent` set's it to 0, does some stuff that if collision gets detected etc. calls `break_block` that sets it to 1, then back in ``apply_movent` if it's set to 1, it plays the sound of breaking a block (VANILLA_SOUND.ENEMIES_BOSS_CAVEMAN_CRUSH)

Reply:
I was using the past-tense of `hit`, so it should mean the same as `post_hit`, but added `post` to make it more obvious.

Updated the comments quite a bit and added the `broke_block`. (I called it `broke_block` to align with the `break_block` virtual function name, and because I forgot about the `destroying_wall` recommendation).